module_utils/network_lsr/argument_validator.py

```diff
@@ -196,7 +196,14 @@ def __init__(  # pylint: disable=too-many-arguments
 
     def _validate_impl(self, value, name):
         if not isinstance(value, Util.STRING_TYPE):
-            raise ValidationError(name, "must be a string but is '%s'" % (value))
+            if name[-3:] == "mac":
```

Comment:
This belongs in an ArgValidatorMac class.

Reply:
I validated in the ArgValidatorMac class as following:
```
        if type(value) == int:
            try:
                value = Util.num_to_mac(value, self.force_len)
            except MyError:
                raise ValidationError(
                    name, "value '%s' is not a valid MAC address" % (value)
                )
        if not isinstance(value, Util.STRING_TYPE):
            raise ValidationError(
                name,
                "must be a string and value should be quoted, but is '%s', " % (value),
            )
```